### DISCUSSION TITLE
May I know what's the new release version and target date to support jakarta namespace change for servlet-api (javax.servlet)

### DISCUSSION THREAD

**Discussion by JasmineXieCCD** [OpenIdentityPlatform/OpenAM#844] [Ideas]
May I know what's the new release version and target date to support jakarta namespace change for servlet-api (javax.servlet)
Such as File:
https://github.com/OpenIdentityPlatform/OpenAM/blob/master/openam-federation/openam-federation-library/src/main/java/org/forgerock/openam/saml2/SAML2ActorFactory.java
https://github.com/OpenIdentityPlatform/OpenAM/blob/master/openam-federation/openam-idpdiscovery/src/main/java/com/sun/identity/saml2/idpdiscovery/ConfiguratorFilter.java
https://github.com/OpenIdentityPlatform/OpenAM/blob/master/openam-shared/src/main/java/com/sun/identity/common/ShutdownServletContextListener.java

**Comment by joelcharlebois** [comment]
Jakarta EE 10 Support - Roadmap or Timelines

I am also interested to know if there is a roadmap or timeline to release a Jakarta EE 10 compliant version of OpenAM.   

The latest Wildfly 27 and JBoss EAP 8.x (currently in Beta) are Jakarta EE 10 compliant and no longer support the older Java EE 8 javax namespaces.  The latest release of OpenAM does not appear to deploy and run correctly on Jakarta EE 10 compliant application servers due to the dependencies on the older Java EE 8 javax namespaces.  

I imagine there may be two major versions of OpenAM supported in parallel for a short while, to allow time for OpenAM users to migrate, with the current major version being Java EE 8 compliant and another being Jakarta EE 10 compliant, and with bug fixes applied to both.  

Thanks.

**Comment by chinkinsei** [comment]
hello, namespace issue maybe handled by migration tools like this, if anyone will give it a try and report back? thanks!

https://tomcat.apache.org/download-migration.cgi

**Comment by OnelSarmiento** [comment]
> hello, namespace issue maybe handled by migration tools like this, if anyone will give it a try and report back? thanks!
> 
> https://tomcat.apache.org/download-migration.cgi

I tried this but I'm having issue with transient dependencies of openam-federation and openam-shared

**Comment by saugion** [comment]
+1, in the last version there are still methods relying on javax

**Comment by sanx72** [comment]
Can we get a response to the original question please? When will there be a version of OpenAM that supports the jakarta.* namespace and what is the timescale? Tomcat 11 no longer supports javax.* namespace so we can't use Apache Felix as a bridge any more.

**Comment by ronperkinsuk** [comment]
**When will OpenAM support the jakarta namespace?**

This is becoming increasingly critical as more application containers now require the use of jakarta.* instead of javax.*.

**Comment by vharseko** [comment]
* [tomcat 9 Supported Java Versions 8+](https://tomcat.apache.org/whichversion.html)
* it is not recommended to share a container between applications
* It is preferable to isolate applications in the docker container

**Reply by joelcharlebois** [reply]
I do not necessarily disagree with you, however a couple of comments / questions:

- Docker on Windows is not ideal and I believe not recommended in production environments, which causes heartburn for the primarily Microsoft IT departments.  Yes they could then alternatively install Tomcat 9 and deploy OpenAM themselves. 

- Tomcat 9.0 is likely going to reach End of Life (EOL), possibly in 2027 based on historical EOL cycles and some comments from Tomcat developers.  It is unclear if there will be a follow-on Tomcat version 9.1 or other, given there are already Tomcat 10 and 11 versions. 

I suppose my own biggest concern, is the Tomcat 9 EOL date, seems to be in the near future.  If that turns out to be the case, and no other Tomcat 9.x versions are released, will the OpenAM team have sufficient time to migrate to the jakarta.* namespace prior to Tomcat 9s EOL date?

**Reply by ronperkinsuk** [reply]
We will be moving to Tomcat 11 and Java 17 as a minimum, so if OpenAM are not going to support the jakarta namespace, then we will have no choice but to look for an alternative to OpenAM.

All recommendations are welcome.

**Reply by joelcharlebois** [reply]
I found a few year old "State of the Cat New!" pdf document online, by Rémy Maucherat, Software engineer at Red Hat.  It appears to indicate that the Tomcat team, at least a few years ago, intended to create Tomcat versions 9.11, 9.12 and 9.13, that are Java 8 EE equivalents to Tomcat 11, 12 and 13, with the javax.* namespace. 

So perhaps based on that, there is no intention for OpenAM to support the jakarta.* namespace...  Regardless, I would like to know if the team has some thoughts on this...

Ron, alternatively you could try to drop in Forgerock's / Ping Identity's commercial paid version of OpenAm.  But I suspect you've already looked at that.

**Reply by ajlugt** [reply]
On the topic of support for the javax.* namespace in Tomcat 9.x, see also a quite recent message [The future of Tomcat 9](https://www.mail-archive.com/announce@tomcat.apache.org/msg00691.html) from one of the Tomcat release managers:

> Tomcat 9 is the last major Tomcat version supporting Java EE. Therefore, the Tomcat community intends to provide support for Tomcat 9 beyond the 10 years for which major Tomcat versions are typically supported.
> 
> Extended support will be provided via a new 9.1.x branch that will be started when 9.0.x reaches end of support.

**Comment by vharseko** [comment]
OpenAM tested on  [java  '8', '11', '17', '21', '24' os: 'ubuntu-latest', 'macos-latest', 'windows-latest'](https://github.com/OpenIdentityPlatform/OpenAM/actions/runs/14416823424) 

* java End of public updates (free) https://en.wikipedia.org/wiki/Java_version_history
* Tomcat 10 don't support java 8 https://tomcat.apache.org/whichversion.html

**Reply by ronperkinsuk** [reply]
We need OpenAM to be compatible with Tomcat 11 and Java 17.

**Comment by vharseko** [comment]
+ @aldaris

**Comment by vharseko** [comment]
We invite you to review 🧐 and test 👻 : 
- commons 3.0.0-SNAPSHOT: https://github.com/OpenIdentityPlatform/commons/pull/132
- OpenDJ 5.0.0-SNAPSHOT: https://github.com/OpenIdentityPlatform/OpenDJ/pull/532 download [OpenDJ 5.0.0-RC1](https://github.com/OpenIdentityPlatform/OpenDJ/releases/tag/5.0.0-RC1)
- OpenAM 16.0.0-SNAPSHOT: https://github.com/OpenIdentityPlatform/OpenAM/pull/889 download [OpenAM 16.0.0-RC1](https://github.com/OpenIdentityPlatform/OpenAM/releases/tag/16.0.0-RC1)

**Reply by joelcharlebois** [reply]
Gents, one other key component we use is the Policy Agent (J2EE, Web, IIS), to enforce URL access policies.  

With respect to compatibility of the various Policy Agents with the JakartaEE version of OpenAM, the Wildly application server has a namespace compatibility flag or property that when enabled, allows Jakarta EE 9+ and JAVA EE 8 endpoints to correctly communicate with each other.  I do not know the scope of this, but brings up a concern of Policy Agent compatibility with the JakartaEE version of OpenAM.

Can you guys verify communication compatibility between the existing policy agents (J2EE, Web and IIS) with the JakartaEE version of OpenAM?  

Thanks.

**Reply by vharseko** [reply]
OpenAM-JEE-Agents: https://github.com/OpenIdentityPlatform/OpenAM-JEE-Agents/pull/35
